modules/wkt/src/lib/parse-wkb.ts

```diff
@@ -116,100 +137,132 @@ function parsePolygon(view, offset, dimension, littleEndian) {
-
-  return {
-    positions: {value: concatenatedPositions, size: dimension}
-  };
+  return concatenateBinaryPointGeometries(binaryPointGeometries, dimension);
 }
 
-function parseMultiLineString(view, offset, dimension, littleEndian) {
+function parseMultiLineString(view, offset, dimension, littleEndian): BinaryLineGeometry {
   const nLines = view.getUint32(offset, littleEndian);
   offset += 4;
 
-  const lines = [];
+  const binaryLineGeometries: BinaryLineGeometry[] = [];
   for (let i = 0; i < nLines; i++) {
     // Byte order for line
     const littleEndianLine = view.getUint8(offset) === 1;
     offset++;
 
     // Assert type LineString
-    assert(
-      view.getUint32(offset, littleEndianLine) % 1000 === 2,
-      'Inner geometries of MultiLineString not of type LineString'
-    );
+    if (view.getUint32(offset, littleEndianLine) % 1000 !== 2) {
+      throw new Error('WKB: Inner geometries of MultiLineString not of type LineString');
+    }
     offset += 4;
 
     const parsed = parseLineString(view, offset, dimension, littleEndianLine);
-    const {positions} = parsed;
     offset = parsed.offset;
-    lines.push(positions.value);
+    binaryLineGeometries.push(parsed.geometry);
   }
 
-  const concatenatedPositions = new Float64Array(concatTypedArrays(lines).buffer);
-  const pathIndices = lines.map((l) => l.length / dimension).map(cumulativeSum(0));
-  pathIndices.unshift(0);
-
-  return {
-    positions: {value: concatenatedPositions, size: dimension},
-    pathIndices: {value: new Uint16Array(pathIndices), size: 1}
-  };
+  return concatenateBinaryLineGeometries(binaryLineGeometries, dimension);
 }
 
-function parseMultiPolygon(view, offset, dimension, littleEndian) {
+function parseMultiPolygon(view, offset, dimension, littleEndian): BinaryPolygonGeometry {
   const nPolygons = view.getUint32(offset, littleEndian);
   offset += 4;
 
-  const polygons = [];
-  const primitivePolygons = [];
+  const binaryPolygonGeometries: BinaryPolygonGeometry[] = [];
   for (let i = 0; i < nPolygons; i++) {
     // Byte order for polygon
     const littleEndianPolygon = view.getUint8(offset) === 1;
     offset++;
 
     // Assert type Polygon
-    assert(
-      view.getUint32(offset, littleEndianPolygon) % 1000 === 3,
-      'Inner geometries of MultiPolygon not of type Polygon'
-    );
+    if (view.getUint32(offset, littleEndianPolygon) % 1000 !== 3) {
+      throw new Error('WKB: Inner geometries of MultiPolygon not of type Polygon');
+    }
     offset += 4;
 
     const parsed = parsePolygon(view, offset, dimension, littleEndianPolygon);
-    const {positions, primitivePolygonIndices} = parsed;
     offset = parsed.offset;
+    binaryPolygonGeometries.push(parsed.geometry);
+  }
+
+  return concatenateBinaryPolygonGeometries(binaryPolygonGeometries, dimension);
+}
+
+// TODO - move to loaders.gl/schema/gis
+
+function concatenateBinaryPointGeometries(
+  binaryPointGeometries: BinaryPointGeometry[],
+  dimension: number
```

Comment:
In the future we might be able to refactor so that `dimension` is unnecessary, since `binaryPointGeometries.positions.size` should be identical to dimension I think.